### PR TITLE
Address coverage and issues

### PR DESCRIPTION
**Pull request by victimsnino**
## Summary by CodeRabbit

- **New Features**
	- Enhanced naming conventions across various classes and files for consistency and clarity.

- **Bug Fixes**
	- None.

- **Refactor**
	- Standardized naming of types and member variables, improving readability and maintainability.
	- Streamlined the implementation of certain functions to reduce overhead.
	- Improved memory management and encapsulation of observer behavior in specific classes.

- **Style**
	- Updated all instances of variables and types to follow lowercase naming conventions where applicable.
	- Renamed thread-local variables in test cases for clarity.

- **Tests**
	- Adjusted test cases to align with new method and signal names following refactoring.
	- Expanded test coverage for error handling and retry mechanisms in observables.

- **Chores**
	- Minor housekeeping changes for improved code clarity without impacting functionality.

### CODE REVIEW

**Comment by victimsnino** [comment]
@coderabbitai full review
@coderabbitai summary

**Comment by victimsnino** [comment]
@coderabbitai full review

**Comment by victimsnino** [comment]
@coderabbitai summary

File: src/rpp/rpp/operators/on_error_resume_next.hpp [hunk 1]
```diff
@@ -17,55 +17,100 @@
...
+            state->observer.on_next(std::forward<T>(v));
         }
 
         void on_error(const std::exception_ptr& err) const
         {
-            std::optional<std::invoke_result_t<Selector, std::exception_ptr>> selector_obs;
             try
             {
-                selector_obs.emplace(selector(err));
+                selector(err).subscribe(on_error_resume_next_inner_observer_strategy<TObserver>{std::shared_ptr<TObserver>(state, &state->observer)});
```
Comment (CorentinBT) [inline src/rpp/rpp/operators/on_error_resume_next.hpp:85]: Do we need the drain approach for this operator too?

Comment (victimsnino) [reply src/rpp/rpp/operators/on_error_resume_next.hpp:85]: Not sure because it applies only once

**Pull request merged**